src/include/pmfs/pmfs.h

```diff
@@ -0,0 +1,182 @@
+/** Allocated maximum OID value */
+#define OID_VALUE       "PMFS_OID_VALUE"
+
+/** Magic Value */
+#define PMFS_SB_MAGIC           0xda05df50da05df50
+/** PMFS SB version value */
+#define PMFS_SB_VERSION         2
+/** PMFS Layout Version Value */
+#define PMFS_LAYOUT_VERSION     2
+/** Array object stripe size for regular files */
+#define PMFS_DEFAULT_CHUNK_SIZE 1048576
+
+/** Number of A-keys for attributes in any object entry */
+#define INODE_AKEYS     8
+#define INODE_AKEY_NAME "PMFS_INODE"
+#define MODE_IDX        0
+#define OID_IDX         (sizeof(mode_t))
+#define ATIME_IDX       (OID_IDX + sizeof(daos_obj_id_t))
+#define MTIME_IDX       (ATIME_IDX + sizeof(time_t))
+#define CTIME_IDX       (MTIME_IDX + sizeof(time_t))
+#define CSIZE_IDX       (CTIME_IDX + sizeof(time_t))
+#define FSIZE_IDX       (CSIZE_IDX + sizeof(daos_size_t))
+#define SYML_IDX        (FSIZE_IDX + sizeof(daos_size_t))
+
+/** OIDs for Superblock and Root objects */
+#define RESERVED_LO     0
+#define SB_HI           0
+#define ROOT_HI         1
+
+/** Max recursion depth for symlinks */
+#define PMFS_MAX_RECURSION 40
+
+/** struct holding attributes for a PMFS container */
+struct  pmfs_attr {
+	/** Optional user ID for DFS container. */
+	uint64_t da_id;
+	/** Default Chunk size for all files in container */
+	daos_size_t da_chunk_size;
+	/** Default Object Class for all objects in the container */
+	daos_oclass_id_t da_oclass_id;
+	/*
+	 * Consistency mode for the PMFS container: PMFS_RELAXED, PMFS_BALANCED.
+	 * If set to 0 or more generally not set to balanced explicitly, relaxed
+	 * mode will be used. In the future, Balanced mode will be the default.
+	 */
+	uint32_t da_mode;
+};
+
+/** object struct that is instantiated for a PMFS open object */
+struct pmfs_obj {
+ 	/* Reference number */
+ 	int                     ref;
+ 	/** DAOS object ID */
+ 	daos_obj_id_t           oid;
+ 	/** mode_t containing permissions & type */
+ 	mode_t                  mode;
+ 	/** open access flags */
+ 	int                     flags;
+ 	/** DAOS object ID of the parent of the object */
+ 	daos_obj_id_t           parent_oid;
+ 	/** entry name of the object in the parent */
+ 	char                    name[PMFS_MAX_NAME + 1];
+ 	/** File size */
+ 	daos_size_t             file_size;
+ 	/** Symlink value if object is a symbolic link */
+ 	char                    *value;
+ 	/** Default chunk size for all entries in dir */
+ 	daos_size_t             chunk_size;
+};
+
+/** pmfs struct that is instantiated for a mounted PMFS namespace */
+struct pmfs {
+ 	/** flag to indicate whether the dfs is mounted */
+ 	bool                    mounted;
+ 	/** flag to indicate whether pmfs is mounted with balanced mode (DTX) */
+ 	bool                    use_dtx;
+ 	/** lock for threadsafety */
+ 	pthread_mutex_t         lock;
+ 	/** uid - inherited from container. */
+ 	uid_t                   uid;
+ 	/** gid - inherited from container. */
+ 	gid_t                   gid;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs